packages/contracts-bedrock/src/dispute/FaultDisputeGame.sol

```diff
@@ -425,8 +438,11 @@ contract FaultDisputeGame is IFaultDisputeGame, Clone, Semver {
             })
         );
 
-        // Set the L1 head hash at the time of the game's creation.
-        l1Head = Hash.wrap(blockhash(block.number - 1));
+        // Set the L1 head hash to the block hash of the L1 block number provided.
+        // This call can revert if the block hash oracle does not have information
+        // about the block number provided to it. This revert will bubble up to the
+        // DisputeGameFactory and prevent the game from being created.
+        l1Head = BLOCK_HASH_ORACLE.load(_getArgUint256(0x40));
```

Comment:
nit: Unless this is much more gas efficient, dedup code here by calling `l1BlockNumber()`